src/main/java/org/aksw/iguana/commons/time/DurationLiteral.java

```diff
@@ -0,0 +1,90 @@
+package org.aksw.iguana.commons.time;
+
+import org.apache.jena.datatypes.DatatypeFormatException;
+import org.apache.jena.datatypes.RDFDatatype;
+import org.apache.jena.graph.impl.LiteralLabel;
+import org.apache.jena.vocabulary.XSD;
+
+import java.time.Duration;
+
+/**
+ * This class is used to convert a Java Duration object to a typed RDF literal. The literal is typed as xsd:duration.<br/>
+ * TODO: This class temporarily fixes an issue with Jena.
+ */
+public class DurationLiteral implements RDFDatatype {
+
+    private final Duration duration;
```

Comment:
Proper implementation of xsd::duration requires a value for days down to milliseconds and a second one for years and months. (because a conversion months<->days is not clearly defined)
(https://www.w3.org/TR/xmlschema11-2/#duration)
Although if you do not need months in your usecase, using xsd:dayTimeDuration (https://www.w3.org/TR/xmlschema11-2/#dayTimeDuration) might be more appropriate.

Reply:
dayTimeDuration sounds like a good fit for our use-case

Reply:
should we change it indefinitely to dayTimeDuration?